rules/rules-reviewed/azure/springboot/spring-boot-to-azure-restricted-config.windup.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ruleset id="spring-boot-to-azure-restricted-config"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset identifies restricted configurations in Spring Boot applications for migration to Azure Spring Apps.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-xml,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="springboot"/>
+        <targetTechnology id="azure-spring-apps"/>
+    </metadata>
+    <rules>
+        <rule id="spring-boot-to-azure-restricted-config-01000">
+            <when>
+                <or>
+                    <filecontent filename="{*}.{extensions}" pattern="{pattern}"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Restricted configurations found" category-id="information" effort="0">
+                    <message>
+                        The application uses restricted configurations for Azure Spring Apps.
+                        These properties are automatically injected into your application environment by Azure Spring Apps to access Config Server and Service Discovery.
+                        Please remove them from your application, including configuration files, config server files, command line parameters, Java system attributes, and environment variables.
+
+                        If configured in **configuration files**: they will be ignored and overrided by Azure Spring Apps.
+                         
+                        If configured in **Config Server files**, **command line parameters**, **Java system attribute**, **environment variable**: they need to be removed or you might experience conflicts and unexpected behavior.
+                    </message>
+                    <link title="Remove restricted configurations" href="http://aka.ms/spring-cloud-to-asa?toc=%2Fazure%2Fspring-apps%2Ftoc.json&amp;bc=%2Fazure%2Fspring-apps%2Fbreadcrumb%2Ftoc.json&amp;pivots=sc-standard-tier#remove-restricted-configurations"/>
```

Comment:
What about using the learn.microsoft.com URL? We are planning to add a campaign ID to all our URLs and this will be easier if they all follow the same format.

 https://learn.microsoft.com/azure/developer/java/migration/migrate-spring-cloud-to-azure-spring-apps?toc=%2Fazure%2Fspring-apps%2Ftoc.json&amp%3Bbc=%2Fazure%2Fspring-apps%2Fbreadcrumb%2Ftoc.json&amp%3Bpivots=sc-standard-tier&pivots=sc-standard-tier#remove-restricted-configurations